pkg/volume/glusterfs/glusterfs.go

```diff
@@ -1048,13 +1056,36 @@ func parseClassParameters(params map[string]string, kubeClient clientset.Interfa
 	return &cfg, nil
 }
 
+// getVolumeID returns volumeID from the PV or volumename.
+func getVolumeID(pv *v1.PersistentVolume, volumeName string) (string, error) {
+	volumeID := ""
+
+	// Get volID from pvspec if available, else fill it from volumename.
+	if pv != nil {
+		if pv.Annotations["VolID"] != "" {
+			volumeID = pv.Annotations["VolID"]
+		} else {
+			volumeID = dstrings.TrimPrefix(volumeName, volPrefix)
```

Comment:
What if there is no prefix? Should it fail?

Reply:
Volumes were always created with prefix. If its changed with custom volume name it should have an entry in pv spec. What I am trying to say here is  it should fall into any of these conditions.

Reply:
How will the error look like when someone modifies the spec not to include `vol_` or remove annotations? `TrimPrefix` won't do anything, it goes to heketi, heketi says this volume does not exist. Is the error message clear enough for the user to spot the error?

Reply:
Yes, in that case heketi returns proper error which states the volume does not exist.